sympy/functions/combinatorial/numbers.py

```diff
@@ -423,18 +423,27 @@ class bernoulli(Function):
     The Bernoulli numbers are a sequence of rational numbers
     defined by `B_0 = 1` and the recursive relation (`n > 0`):
 
-    .. math :: 0 = \sum_{k=0}^n \binom{n+1}{k} B_k
+    .. math :: n+1 = \sum_{k=0}^n \binom{n+1}{k} B_k
 
-    They are also commonly defined by their exponential generating
-    function, which is `\frac{x}{e^x - 1}`. For odd indices > 1, the
-    Bernoulli numbers are zero.
+    This gives `B_1 = +\frac12`, which allows the sequence to be
+    interpolated by an entire function intimately connected to the
+    Riemann zeta function and confers several theoretical advantages [5]_:
+
+    .. math :: B_n = -n\zeta(1-n)
+
+    Older versions of SymPy and some other sources define `B_1 = -\frac12`.
+    The resulting sequence can be obtained with ``(-1)**n*bernoulli(n)``.
```

Comment:
I think it would be good to use Sphinx's `.. versionchanged` at the bottom of the docstring and include a reference to Luschny's Bernoulli manifesto or some other document that discusses the different conventions:
https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-versionchanged